dauphine-open-data-lib/src/main/java/io/github/oliviercailloux/y2018/opendata/entity/Lecture.java

```diff
@@ -1,180 +1,74 @@
 package io.github.oliviercailloux.y2018.opendata.entity;
 
-import java.io.Serializable;
-import java.time.Instant;
-import java.util.Objects;
-import java.util.Optional;
+import java.util.Date;
+
 import javax.persistence.Column;
-import javax.persistence.Embedded;
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import com.google.common.base.Strings;
+import javax.persistence.ManyToOne;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
 
 @Entity
-public class Lecture extends AbstractEntity {
+@NoArgsConstructor
+@RequiredArgsConstructor
+@AllArgsConstructor
+@Data
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.NONE)
+public class Lecture implements io.github.oliviercailloux.y2018.opendata.entity.Entity {
 
 	private static final long serialVersionUID = -6829937183172871605L;
 
-	/**
-	 * The id Lecture Not <code>null</code>.
-	 */
 	@Id
 	@GeneratedValue(strategy = GenerationType.AUTO)
-	private long id;
+	@XmlElement
+	private Long id;
 
 	/**
-	 * Course (basic information about course (author, type ...)
+	 * The associated course 
 	 */
-	@JoinColumn(nullable = false)
+	@NonNull
+	@ManyToOne(optional = false)
+	@XmlElement
 	private Course course;
 
+	@NonNull
+	@Column(nullable = false)
+	@XmlElement
+	private Date date;
+
 	/**
-	 * Date of Lecture
+	 * Duration in minutes
 	 */
-	private Instant date;
-	
-    /**
-     * Date of dur	ation measured in minutes
-     */
+	@NonNull
 	@Column(nullable = false)
-	private int duration;
+	@XmlElement
+	private Integer duration;

```

Comment:
Pourquoi ce choix plutôt que le type primitif utilisé précédemment ? Le type primitif serait mieux ici puisqu’il évite les null.

Reply:
Je suis de manière générale contre les null, en revanche dans un contexte de base de données, je l'accepte et je pense même que c'est nécessaire.
Pour prendre cet attribut comme exemple, si j'avais gardé le type primitif, on n'aurait jamais pu différencier une durée de 0 et une durée non initialisée.

Que ce soit avec les types primitifs ou les String vides, je ne pense pas que les valeurs par défaut soient la bonne méthode pour éviter les valeurs null. Les valeurs par défaut entraînent une perte d'information et peuvent masquer des oublis / erreurs. 

Selon moi, la problématique du null c'est que c'est un pointeur vers "rien" et qu'il ne devrait pas être utiliser pour représenter l'absence de données dans du code.
Il faut donc éviter au possible son utilisation (avec des types option / some of) mais pas pour autant supprimer l'information de présence / absence d'une donnée

Reply:
Certes en général, mais mon propos est issu de l’idée que une Lecture n’est jamais sans duration, et ne devrait donc pas être persistée sans cet élément. Qu’il soit null ou zéro (ou négatif), d’ailleurs.